clash-cores/src/Clash/Cores/EbTb.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE CPP #-}
+
+-- |
+--   Copyright   :  (C) 2024, QBayLogic B.V.
+--   License     :  BSD2 (see the file LICENSE)
+--   Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+--
+--   8b/10b encoding and decoding functions
+module Clash.Cores.EbTb where
+
+import qualified Clash.Cores.EbTb.Decoder as Dec
+import qualified Clash.Cores.EbTb.Encoder as Enc
+import Clash.Prelude
+
+-- | Data type that contains a 'BitVector 8' with the corresponding error
+--   condition of the decode function
+data DataWord
+  = Dw (BitVector 8)
+  | Cw (BitVector 8)
+  | DwError (BitVector 8)
+  | RdError (BitVector 8)
```

Comment:
I guess `RdError` is (Running) Disparity error, but what's DwError?

Reply:
`DwError` is when a 10b code group does not correspond to any data word, which is possible. Is there a way to properly document this?

Reply:
By the way, the reason these still contain `BitVector`s is because @christiaanb suggested to do the decoding on a best-effort basis, to still be able to see what happens when a `DataWord` is not correct.

Reply:
I have added a few lines to explain these different constructors.